folding-schemes/src/folding/hypernova/mod.rs

```diff
@@ -78,8 +81,11 @@ impl<F: PrimeField> Witness<F> {
         // always.
         Self { w, r_w: F::zero() }
     }
-    pub fn dummy(ccs: &CCS<F>) -> Self {
-        Witness::<F>::new(vec![F::zero(); ccs.n - ccs.l - 1])
+}
+
+impl<F: PrimeField> Dummy<&CCS<F>> for Witness<F> {
+    fn dummy(ccs: &CCS<F>) -> Self {
+        Self::new(vec![F::zero(); ccs.n - ccs.l - 1])
```

Comment:
Because `dummy_witness_instance` in turn calls `W::dummy` and `U::dummy` 😂
https://github.com/privacy-scaling-explorations/sonobe/blob/05d9495dd30b3cd7513a57df447f1b1e3f77fc12/folding-schemes/src/arith/mod.rs#L48-L54

For why I didn't implement the creation of dummy values in `dummy_witness_instance`, let me first list all possible combinations of `Arith`, `W` and `U`:
- `CCS`, `hypernova::Witness`, `hypernova::CCCS`
- `CCS`, `hypernova::Witness`, `hypernova::LCCCS`
- `R1CS`, `nova::Witness`, `nova::CommittedInstance`
- `R1CS`, `protogalaxy::Witness`, `protogalaxy::CommittedInstance<TRUE>`,
- `R1CS`, `protogalaxy::Witness`, `protogalaxy::CommittedInstance<FALSE>`

As we can see, the same witness type may be associated with multiple committed instances. Thus, if we create dummy values directly in `Arith::dummy_witness_instance`, there would be some duplicated code for building dummy `W`.

Another advantage of `Dummy` is that we can create dummy values without knowing the instance of `Arith`, while `Arith::dummy_witness_instance` requires `Dummy::Cfg` to be a reference to `Arith`.
For instance, we can create a dummy committed instance in Nova even if we only know `io_len`.
https://github.com/privacy-scaling-explorations/sonobe/blob/05d9495dd30b3cd7513a57df447f1b1e3f77fc12/folding-schemes/src/folding/nova/mod.rs#L77-L86
This is useful for example in the code below, where we don't have to pass `cf_r1cs` to the circuit.
https://github.com/privacy-scaling-explorations/sonobe/blob/05d9495dd30b3cd7513a57df447f1b1e3f77fc12/folding-schemes/src/folding/nova/circuits.rs#L341

Reply:
Ohh dammit!! It's true my bad.. Apologies for the noise. Totally agree with you!